### DISCUSSION TITLE
Button Callback function with arguments auto run

### DISCUSSION THREAD

**Discussion by wsimm** [robotools/vanilla#167] [General]
Hi,
when I create a button with a callback function with arguments it auto runs the function assigned 

`from vanilla import Window, Button

class ButtonDemo:

     def __init__(self):
         self.w = Window((100, 40))
         self.w.button = Button((10, 10, -10, 20), "A Button",
                            callback=self.buttonCallback('somedata'))
         self.w.open()

     def buttonCallback(self, sender):
         print("button hit!")

ButtonDemo()`

**Comment by justvanrossum** [comment]
You have to assign the callback without the `('somedata')` bit (which is calling it on the spot).

```python
from vanilla import Window, Button

class ButtonDemo:

 def __init__(self):
     self.w = Window((100, 40))
     self.w.button = Button((10, 10, -10, 20), "A Button",
                        callback=self.buttonCallback)
     self.w.open()

 def buttonCallback(self, sender):
     print("button hit!")

ButtonDemo()
```

**Comment by wsimm** [comment]
how to pass some data through the arguments ?

**Comment by justvanrossum** [comment]
The callback is called by vanilla, and it passes the "sender" argument. You don't pass it arguments yourself.

What are you trying to do?
